app/schemas.py

```diff
@@ -1,9 +1,61 @@
+import re
+from datetime import date
+from datetime import datetime
+
 from pydantic import BaseModel
 from pydantic import ConfigDict
+from pydantic import Field
+from pydantic import field_serializer
+from pydantic import field_validator
+from sqlalchemy_utils import Currency
+
+from app.enums import PriceLocationOSMType
 
 
 class UserBase(BaseModel):
     model_config = ConfigDict(from_attributes=True)
 
     user_id: str
     token: str
+
+
+class PriceCreate(BaseModel):
+    model_config = ConfigDict(from_attributes=True, arbitrary_types_allowed=True)
+
+    product_code: str = Field(min_length=1)
+    price: float
+    currency: str | Currency
+    location_osm_id: int
+    location_osm_type: PriceLocationOSMType
+    date: date
+
+    @field_validator("product_code")
+    def product_code_must_be_only_numbers(cls, v):
+        regex = r"^[0-9]+$"
+        if not re.match(regex, v):
+            raise ValueError("must only contain numbers")
+        return v
+
+    @field_validator("currency")
+    def currency_is_valid(cls, v):
+        try:
+            return Currency(v).code
+        except ValueError:
+            raise ValueError("not a valid currency code")
+
+    @field_validator("location_osm_id")
+    def location_osm_id_must_be_positive(cls, v):
```

Comment:
Same thing here, you can directly require it to be positive in the Field:
https://docs.pydantic.dev/latest/concepts/fields/#numeric-constraints

Reply:
top :raised_hands: :heavy_check_mark: